code/src/components/App.jsx

```diff
@@ -771,6 +775,9 @@ export default class App extends React.Component {
         onLayerSelect={this.onLayerSelect}
       />
     } else {
+      mapProps.mapStyle.layers = this.addDefaultOrdinalFilter(mapProps.mapStyle.layers);
+      mapProps.selectableLayers = this.addDefaultOrdinalFilter(mapProps.selectableLayers);
```

Comment:
I'm a little bit concerned about perf here, that it will iterate on both array on each change. Whilst selectableLayers seems to be usually small, mapStyle.layers may be quite bit. Would love to improve this but don't see how.

Reply:
this looks much nicer but essentially a different way of doing the same thing. Given the looming release I'd prefer to keep changes as small as possible. So then will probably keep it as is for now and will hopefully change this later.

Reply:
It all depends on how frequent is setStyle, I would say